universal-login-wallet/src/ui/react/Home/HomeScreen.tsx

```diff
@@ -24,42 +26,42 @@ const HomeScreen = () => {
             <Notice message={notice} />
             <p className="dashboard-ens-name">{walletPresenter.getName()}</p>
             <Switch>
-              <Route path="/" exact>
+              <Route path={basePath} exact>
                 <Funds
                   deployedWallet={walletService.getDeployedWallet()}
-                  onTopUpClick={() => history.push('/topUp')}
-                  onSendClick={() => history.push('/transfer')}
-                  onDeviceMessageClick={() => history.push('/devices/approveDevice')}
+                  onTopUpClick={() => history.push(join(basePath, 'topUp'))}
+                  onSendClick={() => history.push(join(basePath, 'transfer'))}
+                  onDeviceMessageClick={() => history.push(join(basePath, 'devices', 'approveDevice'))}
                   className="jarvis-styles"
                 />
               </Route>
-              <Route path="/devices">
+              <Route path={join(basePath, 'devices')}>
                 <Devices
                   walletService={walletService}
                   onAccountDisconnected={() => history.push('/welcome')}
-                  basePath="/devices"
+                  basePath={join(basePath, 'devices')}
                   className="jarvis-styles"
                 />
               </Route>
-              <Route path="/backup">
+              <Route path={join(basePath, 'backup')}>
                 <BackupCodes
                   deployedWallet={walletService.getDeployedWallet()}
                   className="jarvis-backup"
                 />
               </Route>
-              <Route path="/topUp">
+              <Route path={join(basePath, 'topUp')}>
```

Comment:
basePath => currentPath
extract `pathFor(currentPath, 'topup')`, that uses `replaceSecondSegment` or similar